### PR TITLE
mautrix-signal: 0.4.3 -> unstable-2023-12-30

### LINKED ISSUE #248379: mautrix-signal crashes at startup
### Describe the bug
`mautrix-signal` crashes during startup with:
```
Aug 10 15:13:58 matrix systemd[1]: mautrix-signal.service: Scheduled restart job, restart counter is at 39324.
Aug 10 15:13:58 matrix systemd[1]: Stopped Synapse <-> Signal puppeting bridge (mautrix-signal).
Aug 10 15:13:58 matrix systemd[1]: mautrix-signal.service: Consumed 3.405s CPU time, no IP traffic.
Aug 10 15:13:58 matrix systemd[1]: Starting Synapse <-> Signal puppeting bridge (mautrix-signal)...
Aug 10 15:13:58 matrix systemd[1]: Started Synapse <-> Signal puppeting bridge (mautrix-signal).
Aug 10 15:14:01 matrix mautrix-signal[565521]: [2023-08-10 15:14:01,588] [INFO@mau.init] Initializing mautrix-signal 0.4.3+dev.unknown
Aug 10 15:14:01 matrix mautrix-signal[565521]: [2023-08-10 15:14:01,596] [CRITICAL@mau.init] Unexpected error in initialization
Aug 10 15:14:01 matrix mautrix-signal[565521]: Traceback (most recent call last):
Aug 10 15:14:01 matrix mautrix-signal[565521]:   File "/nix/store/ibd336k0716m02yk1glw418csccdpm8c-python3.10-mautrix-0.20.0/lib/python3.10/site-packages/mautrix/util/program.py", line 101, in _prepare
Aug 10 15:14:01 matrix mautrix-signal[565521]:     self.prepare()
Aug 10 15:14:01 matrix mautrix-signal[565521]:   File "/nix/store/ibd336k0716m02yk1glw418csccdpm8c-python3.10-mautrix-0.20.0/lib/python3.10/site-packages/mautrix/bridge/bridge.py", line 133, in prepare
Aug 10 15:14:01 matrix mautrix-signal[565521]:     self.prepare_bridge()
Aug 10 15:14:01 matrix mautrix-signal[565521]:   File "/nix/store/dp8miy3x2c212lbjw591xkqvkl5r7qxq-python3.10-mautrix-signal-0.4.3/lib/python3.10/site-packages/mautrix_signal/__main__.py", line 64, in prepare_bridge
Aug 10 15:14:01 matrix mautrix-signal[565521]:     super().prepare_bridge()
Aug 10 15:14:01 matrix mautrix-signal[565521]:   File "/nix/store/ibd336k0716m02yk1glw418csccdpm8c-python3.10-mautrix-0.20.0/lib/python3.10/site-packages/mautrix/bridge/bridge.py", line 202, in prepare_bridge
Aug 10 15:14:01 matrix mautrix-signal[565521]:     self.matrix = self.matrix_class(bridge=self)
Aug 10 15:14:01 matrix mautrix-signal[565521]:   File "/nix/store/dp8miy3x2c212lbjw591xkqvkl5r7qxq-python3.10-mautrix-signal-0.4.3/lib/python3.10/site-packages/mautrix_signal/matrix.py", line 57, in __init__
Aug 10 15:14:01 matrix mautrix-signal[565521]:     super().__init__(bridge=bridge)
Aug 10 15:14:01 matrix mautrix-signal[565521]:   File "/nix/store/ibd336k0716m02yk1glw418csccdpm8c-python3.10-mautrix-0.20.0/lib/python3.10/site-packages/mautrix/bridge/matrix.py", line 182, in __init__
Aug 10 15:14:01 matrix mautrix-signal[565521]:     self.e2ee = EncryptionManager(
Aug 10 15:14:01 matrix mautrix-signal[565521]:   File "/nix/store/ibd336k0716m02yk1glw418csccdpm8c-python3.10-mautrix-0.20.0/lib/python3.10/site-packages/mautrix/bridge/e2ee.py", line 128, in __init__
Aug 10 15:14:01 matrix mautrix-signal[565521]:     self.delete_outdated_inbound = del_cfg["delete_outdated_inbound"]
Aug 10 15:14:01 matrix mautrix-signal[565521]:   File "/nix/store/g0p2myfv593hg4lskxnk5nrk92xiqjis-python3.10-ruamel-yaml-0.17.21/lib/python3.10/site-packages/ruamel/yaml/comments.py", line 927, in __getitem__
Aug 10 15:14:01 matrix mautrix-signal[565521]:     return ordereddict.__getitem__(self, key)
Aug 10 15:14:01 matrix mautrix-signal[565521]: KeyError: 'delete_outdated_inbound'
Aug 10 15:14:02 matrix systemd[1]: mautrix-signal.service: Main process exited, code=exited, status=1/FAILURE
Aug 10 15:14:02 matrix systemd[1]: mautrix-signal.service: Failed with result 'exit-code'.
Aug 10 15:14:02 matrix systemd[1]: mautrix-signal.service: Consumed 3.398s CPU time, no IP traffic.
Aug 10 15:14:05 matrix systemd[1]: mautrix-signal.service: Scheduled restart job, restart counter is at 39325.
Aug 10 15:14:05 matrix systemd[1]: Stopped Synapse <-> Signal puppeting bridge (mautrix-signal).
```

Judging by this upstream issue https://github.com/mautrix/python/issues/154 which has the same stack trace, the crash is caused by using an incompatible version of the `mautrix` dependency with `mautrix-signal`. [`requirements.txt` in `mautrix-signal`'s repo](https://github.com/mautrix/signal/blob/64bfed8f05e1e30edd719f2f9a33098ee89229d8/requirements.txt) specifies `mautrix>=0.19.14,<0.20` but currently in nixpkgs it's provided with `mautrix-0.20.0` as a dependency. It seems that an older `mautrix` version should be used to fix the crash.

### Steps To Reproduce
1. It might somehow be specific to my `config.yaml` (judging by the stack trace, it might be related to e2ee), so here's mine:
    ```yaml
    # Homeserver details
    homeserver:
        # The address that this appservice can use to connect to the homeserver.
        address: http://127.0.0.1:8008
        # The domain of the homeserver (also known as server_name, used for MXIDs, etc).
        domain: <redacted>
        # Whether or not to verify the SSL certificate of the homeserver.
        # Only applies if address starts with https://
        verify_ssl: true
        # What software is the homeserver running?
        # Standard Matrix homeservers like Synapse, Dendrite and Conduit should just use "standard" here.
        software: standard
        # Number of retries for all HTTP requests if the homeserver isn't reachable.
        http_retry_count: 4
        # The URL to push real-time bridge status to.
        # If set, the bridge will make POST requests to this URL whenever a user's Signal connection state changes.
        # The bridge will use the appservice as_token to authorize requests.
        status_endpoint:
        # Endpoint for reporting per-message status.
        message_send_checkpoint_endpoint:
        # Maximum number of simultaneous HTTP connections to the homeserver.
        connection_limit: 100
        # Whether asynchronous uploads via MSC2246 should be enabled for media.
        # Requires a media repo that supports MSC2246.
        async_media: false
    
    # Application service host/registration related details
    # Changing these values requires regeneration of the registration.
    appservice:
        # The address that the homeserver can use to connect to this appservice.
        address: http://localhost:29328
        # When using https:// the TLS certificate and key files for the address.
        tls_cert: false
        tls_key: false
    
        # The hostname and port where this appservice should listen.
        hostname: 0.0.0.0
        port: 29328
        # The maximum body size of appservice API requests (from the homeserver) in mebibytes
        # Usually 1 is enough, but on high-traffic bridges you might need to increase this to avoid 413s
        max_body_size: 1
    
        # The full URI to the database. SQLite and Postgres are supported.
        # Format examples:
        #   SQLite:   sqlite:///filename.db
        #   Postgres: postgres://username:password@hostname/dbname
        database: postgres:///mautrix-signal?host=/var/run/postgresql
        # Additional arguments for asyncpg.create_pool() or sqlite3.connect()
        # https://magicstack.github.io/asyncpg/current/api/index.html#asyncpg.pool.create_pool
        # https://docs.python.org/3/library/sqlite3.html#sqlite3.connect
        # For sqlite, min_size is used as the connection thread pool size and max_size is ignored.
        # Additionally, SQLite supports init_commands as an array of SQL queries to run on connect (e.g. to set PRAGMAs).
        database_opts:
            min_size: 1
            max_size: 10
        id: signal
        # Username of the appservice bot.
        bot_username: signalbot
        # Display name and avatar for bot. Set to "remove" to remove display name/avatar, leave empty
        # to leave display name/avatar as-is.
        bot_displayname: Signal bridge bot
        bot_avatar: mxc://maunium.net/wPJgTQbZOtpBFmDNkiNEMDUp
    
        # Whether or not to receive ephemeral events via appservice transactions.
        # Requires MSC2409 support (i.e. Synapse 1.22+).
        # You should disable bridge -> sync_with_custom_puppets when this is enabled.
        ephemeral_events: true
    
        # Authentication tokens for AS <-> HS communication. Autogenerated; do not modify.
        as_token: <redacted>
        hs_token: <redacted>
    
    # Prometheus telemetry config. Requires prometheus-client to be installed.
    metrics:
        enabled: false
        listen_port: 8000
    
    # Manhole config.
    manhole:
        # Whether or not opening the manhole is allowed.
        enabled: false
        # The path for the unix socket.
        path: /var/tmp/mautrix-signal.manhole
        # The list of UIDs who can be added to the whitelist.
        # If empty, any UIDs can be specified in the open-manhole command.
        whitelist:
        - 0
    signal:
        # Path to signald unix socket
        socket_path: /run/signald/signald.sock
        # Directory for temp files when sending files to Signal. This should be an
        # absolute path that signald can read. For attachments in the other direction,
        # make sure signald is configured to use an absolute path as the data directory.
        outgoing_attachment_dir: /var/lib/mautrix-signal/outgoing-attachments
        # Directory where signald stores avatars for groups.
        avatar_dir: /var/lib/signald/avatars
        # Directory where signald stores auth data. Used to delete data when logging out.
        data_dir: /var/lib/signald/data
        # Whether or not unknown signald accounts should be deleted when the bridge is started.
        # When this is enabled, any UserInUse errors should be resolved by restarting the bridge.
        delete_unknown_accounts_on_start: false
        # Whether or not message attachments should be removed from disk after they're bridged.
        remove_file_after_handling: true
        # Whether or not users can register a primary device
        registration_enabled: true
        # Whether or not to enable disappearing messages in groups. If enabled, then the expiration
        # time of the messages will be determined by the first users to read the message, rather
        # than individually. If the bridge has a single user, this can be turned on safely.
        enable_disappearing_messages_in_groups: true
    
    # Bridge config
    bridge:
        # Localpart template of MXIDs for Signal users.
        # {userid} is replaced with the UUID of the Signal user.
        username_template: signal_{userid}
        # Displayname template for Signal users.
        # {displayname} is replaced with the displayname of the Signal user, which is the first
        # available variable in displayname_preference. The variables in displayname_preference
        # can also be used here directly.
        displayname_template: '{displayname} (Signal)'
        # Whether or not contact list displaynames should be used.
        # Possible values: disallow, allow, prefer
        #
        # Multi-user instances are recommended to disallow contact list names, as otherwise there can
        # be conflicts between names from different users' contact lists.
        contact_list_names: disallow
        # Available variables: full_name, first_name, last_name, phone, uuid
        displayname_preference:
        - full_name
        - phone
        autocreate_group_portal: true
        # Whether or not to create portals for all contacts on login/connect.
        autocreate_contact_portal: false
        # Whether or not to make portals of Signal groups in which joining via invite link does
        # not need to be approved by an administrator publicly joinable on Matrix.
        public_portals: false
        # Whether or not to use /sync to get read receipts and typing notifications
        # when double puppeting is enabled
        sync_with_custom_puppets: false
        # Whether or not to update the m.direct account data event when double puppeting is enabled.
        # Note that updating the m.direct event is not atomic (except with mautrix-asmux)
        # and is therefore prone to race conditions.
        sync_direct_chat_list: false
        # Allow using double puppeting from any server with a valid client .well-known file.
        double_puppet_allow_discovery: false
        # Servers to allow double puppeting from, even if double_puppet_allow_discovery is false.
        double_puppet_server_map:
            example.com: https://example.com
        login_shared_secret_map:
            example.com: foo
        federate_rooms: false
        # End-to-bridge encryption support options.
        #
        # See https://docs.mau.fi/bridges/general/end-to-bridge-encryption.html for more info.
        encryption:
            # Allow encryption, work in group chat rooms with e2ee enabled
            allow: true
            # Default to encryption, force-enable encryption in all portals the bridge creates
            # This will cause the bridge bot to be in private chats for the encryption to work properly.
            default: true
            # Whether to use MSC2409/MSC3202 instead of /sync long polling for receiving encryption-related data.
            appservice: false
            # Require encryption, drop any unencrypted messages.
            require: false
            # Enable key sharing? If enabled, key requests for rooms where users are in will be fulfilled.
            # You must use a client that supports requesting keys from other users to use this feature.
            allow_key_sharing: false
            # Options for deleting megolm sessions from the bridge.
            delete_keys:
                # Beeper-specific: delete outbound sessions when hungryserv confirms
                # that the user has uploaded the key to key backup.
                delete_outbound_on_ack: false
                # Don't store outbound sessions in the inbound table.
                dont_store_outbound: false
                # Ratchet megolm sessions forward after decrypting messages.
                ratchet_on_decrypt: false
                # Delete fully used keys (index >= max_messages) after decrypting messages.
                delete_fully_used_on_decrypt: false
                # Delete previous megolm sessions from same device when receiving a new one.
                delete_prev_on_new_session: false
                # Delete megolm sessions received from a device when the device is deleted.
                delete_on_device_delete: false
                # Periodically delete megolm sessions when 2x max_age has passed since receiving the session.
                periodically_delete_expired: false
            # What level of device verification should be required from users?
            #
            # Valid levels:
            #   unverified - Send keys to all device in the room.
            #   cross-signed-untrusted - Require valid cross-signing, but trust all cross-signing keys.
            #   cross-signed-tofu - Require valid cross-signing, trust cross-signing keys on first use (and reject changes).
            #   cross-signed-verified - Require valid cross-signing, plus a valid user signature from the bridge bot.
            #                           Note that creating user signatures from the bridge bot is not currently possible.
            #   verified - Require manual per-device verification
            #              (currently only possible by modifying the `trust` column in the `crypto_device` database table).
            verification_levels:
                # Minimum level for which the bridge should send keys to when bridging messages from Telegram to Matrix.
                receive: unverified
                # Minimum level that the bridge should accept for incoming Matrix messages.
                send: unverified
                # Minimum level that the bridge should require for accepting key requests.
                share: cross-signed-tofu
            # Options for Megolm room key rotation. These options allow you to
            # configure the m.room.encryption event content. See:
            # https://spec.matrix.org/v1.3/client-server-api/#mroomencryption for
            # more information about that event.
            rotation:
                # Enable custom Megolm room key rotation settings. Note that these
                # settings will only apply to rooms created after this option is
                # set.
                enable_custom: false
                # The maximum number of milliseconds a session should be used
                # before changing it. The Matrix spec recommends 604800000 (a week)
                # as the default.
                milliseconds: 604800000
                # The maximum number of messages that should be sent with a given a
                # session before changing it. The Matrix spec recommends 100 as the
                # default.
                messages: 100
    
        # Whether to explicitly set the avatar and room name for private chat portal rooms.
        # If set to `default`, this will be enabled in encrypted rooms and disabled in unencrypted rooms.
        # If set to `always`, all DM rooms will have explicit names and avatars set.
        # If set to `never`, DM rooms will never have names and avatars set.
        private_chat_portal_meta: default
        # Whether or not the bridge should send a read receipt from the bridge bot when a message has
        # been sent to Signal. This let's you check manually whether the bridge is receiving your
        # messages.
        # Note that this is not related to Signal delivery receipts.
        delivery_receipts: false
        # Whether or not delivery errors should be reported as messages in the Matrix room.
        delivery_error_reports: true
        # Whether the bridge should send the message status as a custom com.beeper.message_send_status event.
        message_status_events: false
        # Set this to true to tell the bridge to re-send m.bridge events to all rooms on the next run.
        # This field will automatically be changed back to false after it,
        # except if the config file is not writable.
        resend_bridge_info: false
        # Interval at which to resync contacts (in seconds).
        periodic_sync: 0
        # Should leaving the room on Matrix make the user leave on Signal?
        bridge_matrix_leave: true
        # Should the bridge auto-create a group chat on Signal when a ghost is invited to a room?
        # Requires the user to have sufficient power level and double puppeting enabled.
        create_group_on_invite: true
        hacky_contact_name_mixup_detection: false
    
        # Provisioning API part of the web server for automated portal creation and fetching information.
        # Used by things like mautrix-manager (https://github.com/tulir/mautrix-manager).
        provisioning:
            # Whether or not the provisioning API should be enabled.
            enabled: true
            # The prefix to use in the provisioning API endpoints.
            prefix: /_matrix/provision
            # The shared secret to authorize users of the API.
            # Set to "generate" to generate and save a new token.
            shared_secret: <redacted>
            # Segment API key to enable analytics tracking for web server
            # endpoints. Set to null to disable.
            # Currently the only events are login start, QR code scan, and login
            # success/failure.
            segment_key:
            # Optional user_id to use when sending Segment events. If null, defaults to using mxID.
            segment_user_id:
    
        # The prefix for commands. Only required in non-management rooms.
        command_prefix: '!signal'
    
        # Messages sent upon joining a management room.
        # Markdown is supported. The defaults are listed below.
        management_room_text:
            # Sent when joining a room.
            welcome: Hello, I'm a Signal bridge bot.
            # Sent when joining a management room and the user is already logged in.
            welcome_connected: Use `help` for help.
            # Sent when joining a management room and the user is not logged in.
            welcome_unconnected: Use `help` for help or `link` to log in.
            # Optional extra text sent when joining a management room.
            additional_help: ''
    
        # Send each message separately (for readability in some clients)
        management_room_multiple_messages: false
    
        # Permissions for using the bridge.
        # Permitted values:
        #      relay - Allowed to be relayed through the bridge, no access to commands.
        #       user - Use the bridge with puppeting.
        #      admin - Use and administrate the bridge.
        # Permitted keys:
        #        * - All Matrix users
        #   domain - All users on that homeserver
        #     mxid - Specific user
        permissions:
            '<redacted>': admin
        relay:
            # Whether relay mode should be allowed. If allowed, `!signal set-relay` can be used to turn any
            # authenticated user into a relaybot for that chat.
            enabled: false
            # The formats to use when sending messages to Signal via a relay user.
            #
            # Available variables:
            #   $sender_displayname - The display name of the sender (e.g. Example User)
            #   $sender_username    - The username (Matrix ID localpart) of the sender (e.g. exampleuser)
            #   $sender_mxid        - The Matrix ID of the sender (e.g. @exampleuser:example.com)
            #   $message            - The message content
            message_formats:
                m.text: '$sender_displayname: $message'
                m.notice: '$sender_displayname: $message'
                m.emote: '* $sender_displayname $message'
                m.file: $sender_displayname sent a file
                m.image: $sender_displayname sent an image
                m.audio: $sender_displayname sent an audio file
                m.video: $sender_displayname sent a video
                m.location: $sender_displayname sent a location
            relaybot: '@relaybot:example.com'
            # Whether or not invites from non-logged-in users should be relayed
            invite: true
    
        # Format for generating URLs from location messages for sending to Signal
        # Google Maps: 'https://www.google.com/maps/place/{lat},{long}'
        # OpenStreetMap: 'https://www.openstreetmap.org/?mlat={lat}&mlon={long}'
        location_format: https://www.google.com/maps/place/{lat},{long}
    
    # Python logging configuration.
    #
    # See section 16.7.2 of the Python documentation for more info:
    # https://docs.python.org/3.6/library/logging.config.html#configuration-dictionary-schema
    logging:
        version: 1
        formatters:
            colored:
                (): mautrix_signal.util.ColorFormatter
                format: '[%(asctime)s] [%(levelname)s@%(name)s] %(message)s'
            normal:
                format: '[%(asctime)s] [%(levelname)s@%(name)s] %(message)s'
        handlers:
            file:
                class: logging.handlers.RotatingFileHandler
                formatter: normal
                filename: ./mautrix-signal.log
                maxBytes: 10485760
                backupCount: 10
            console:
                class: logging.StreamHandler
                formatter: colored
        loggers:
            mau:
                level: DEBUG
            aiohttp:
                level: INFO
        root:
            level: DEBUG
            # handlers: [file, console]
            handlers: [console]


    ```
2. Fill in `<redacted>` parts in the provided `config.yaml`
3. Start mautrix-signal `mautrix-signal -c config.yaml`

### Expected behavior
No crash

### Notify maintainers
@expipiplus1 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-linux"`
 - host os: `Linux 5.19.9, NixOS, 23.11 (Tapir)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - channels(root): `"nixos-22.05pre335103.6daa4a5c045"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by niklaskorz**
## Description of changes

mautrix-signal's latest released version v0.4.3 still uses the Python codebase
which is broken for new devices, see https://github.com/mautrix/signal/issues/388.
The [new Go version](https://github.com/mautrix/signal/issues/372) fixes this by using the official libsignal as a library and
can be upgraded to directly from the Python version.

Fixes #248379.

If anyone wants to try this out in combination with https://github.com/NixOS/nixpkgs/pull/277368, I also created a combined branch of these changes based on nixos-23.11: `github:niklaskorz/nixpkgs/nixos-23.11-mautrix-signal`.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/li/libsignal-ffi/package.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{ lib, stdenv, fetchFromGitHub, rustPlatform, runCommand, xcodebuild, protobuf, boringssl }:
+let
+  # boring-sys expects the static libraries in build/ instead of lib/
+  boringssl-wrapper = runCommand "boringssl-wrapper" { } ''
+    mkdir $out
+    cd $out
+    cp -r ${boringssl.out}/lib build
+    cp -r ${boringssl.dev}/include include
+  '';
```
Comment (alyaeanyx) [inline pkgs/by-name/li/libsignal-ffi/package.nix:9]: Could we also do this with `ln -s` instead of `cp -r`? This might save some disk space.

File: pkgs/by-name/li/libsignal-ffi/package.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/lib
+    mv target/release/libsignal_ffi.a $out/lib/
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "libsignal-ffi is a C ABI library which exposes Signal protocol logic to languages which can consume a C ABI, such as Swift.";
```
Comment (alyaeanyx) [inline pkgs/by-name/li/libsignal-ffi/package.nix:50]: Shouldn't include the name of the package and the colon at the end ;) (see https://nixos.org/manual/nixpkgs/stable/#var-meta-description)

**Review by cyclic-pentane — COMMENTED** [review]
Nice work, thanks!

**Comment by Ma27** [comment]
Nice! Since I'm currently locked out of my bridge because of the 405 error, I'll happily test this soonish! :)

File: pkgs/by-name/li/libsignal-ffi/package.nix (hunk 1 above)
Comment (niklaskorz) [reply pkgs/by-name/li/libsignal-ffi/package.nix]: Done, looks like boring-sys is fine with symlinks :)

**Review by cyclic-pentane — APPROVED** [review]
LGTM

File: pkgs/by-name/li/libsignal-ffi/package.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "boring-3.1.0" = "sha256-R6hh4K57mgV10nuVcMZETvxlQsMsmGapgCQ7pjuognk=";
+      "curve25519-dalek-4.1.1" = "sha256-p9Vx0lAaYILypsI4/RVsHZLOqZKaa4Wvf7DanLA38pc=";
+    };
+  };
+
+  buildPhase = ''
+    runHook preBuild
+    cargo build --release -p libsignal-ffi
```
Comment (Ma27) [inline pkgs/by-name/li/libsignal-ffi/package.nix:38]: Rather than overriding the `buildPhase` entirely, this can be done with `cargoBuildFlags = [ "-p" "libsignal-ffi" ];`.

File: pkgs/by-name/li/libsignal-ffi/package.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/lib
+    mv target/release/libsignal_ffi.a $out/lib/
```
Comment (Ma27) [inline pkgs/by-name/li/libsignal-ffi/package.nix:45]: `buildRustPackage` should be able to do that on its own actually.

**Review by Ma27 — COMMENTED** [review]
Just two small things I noticed. Going to try it out now and hopefully I can report back soon :)

**Comment by Ma27** [comment]
Bridge is working fine btw :+1:

**Comment by niklaskorz** [comment]
I'm a bit unsure whether `libsignal-ffi` should be its own package or be part of the `mautrix-signal` package. Right now, this PR adds `libsignal-ffi` at version 0.36.1 while the latest version is 0.37.0. I tried running `mautrix-signal` with 0.37.0, but this leads to errors, so the version of `libsignal-ffi` must match unfortunately: https://github.com/mautrix/signal/issues/401

**Review by Ma27 — APPROVED** [review]

**Comment by Ma27** [comment]
> I'm a bit unsure whether libsignal-ffi should be its own package or be part of the mautrix-signal package

At the end of the day it's our job to provide a package set that works _together_, not a bunch of packages at the latest version (of course, that's not a hard rule, but something we should be keeping in mind generally).

My suggestion would be to still leave a warning in the `libsignal-ffi` nix expression, but as amintainer you should be pinged anyways on updates.

While you're at it, feel free to also add me to the list of maintainers for mautrix-signal :)

**Comment by niklaskorz** [comment]
@Ma27 alright, done :)

@expipiplus1 any remarks from your side? do you still want to maintain the mautrix-signal package?

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -13246,6 +13246,13 @@
     githubId = 6391776;
     name = "Nikita Voloboev";
   };
+  niklaskorz = {
+    name = "Niklas Korz";
+    email = "niklas@niklaskorz.de";
+    matrix = "@niklaskorz:korz.dev";
+    github = "niklaskorz";
+    githubId = 590517;
+  };
```
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:13255]: this should be it's own commit called: `maintainers: add niklaskorz`

**Review by Janik-Haag — COMMENTED** [review]
didin't review the package only the maintainers part. Btw please clean up your commit history a bit. the `libsignal-ffi` changes can probably be squashed and the mautrix-signal changes should be squashed into one commit. nixpkgs tries to have a atomic commit history.

**Review by Ma27 — APPROVED** [review]
